Add gettext translations

Snunesnk opened the pull request:
## Issue reference

- Implements #45 

## Changes

- Added `gettext ~> 1.0` as dependency
- Added a dedicated backend (`Lotus.Web.Gettext`) with `default_domain: "lotus"`, 
- Wrapped all user-facing strings (editor, results, widgets, schema explorer, shortcut modal, etc.) with `gettext`/`ngettext`
- Added section in README on how to integrate translation from host app

## Screenshots / Logs (optional)

N/A

## Breaking changes

- Expects `gettext ~> 1.0`

## Documentation updates

- Added section in README about how to add translation for lotus interface, within host app.

## Checklist

- [x] I have linked a GitHub issue using "Closes/Fixes/Resolves #<id>" (required)
- [ ] I have followed the project's coding standards (please let us know if you aren't sure, we're happy to clarify!)
- [ ] Tests added or updated to cover changes
- [x] All tests pass locally (mix test)
- [x] Code formatted (mix format)
- [x] Credo is clean (mix credo)
- [x] Dialyzer is clean (mix dialyzer)
- [ ] CHANGELOG.md updated, if user-facing behavior changed

## Notes for reviewers (optional)

Thank you for your time and consideration :pray:

rodloboz commented on lib/lotus/web/components/layouts.ex:
```diff
@@ -29,7 +29,7 @@ defmodule Lotus.Web.Layouts do
   def logo(assigns) do
     ~H"""
-    <a href={lotus_path("")} title="Lotus Web">
+    <a href={lotus_path("")} title={gettext("Lotus Web")}>
```
This is the brand and there’s no plan to offer Lotus as a white label, so I dont see a good reason to translate this string.

tugayac commented on README.md:
```diff
@@ -301,6 +302,33 @@ lotus_dashboard "/lotus",
+### Providing translations from a host application
+
+1. Create a place for your translations (for example `priv/lotus_gettext` in your host app) and copy the template from `deps/lotus_web/priv/gettext/lotus.pot` as a starting point.
+2. Add locale-specific PO files such as `priv/lotus_gettext/fr/LC_MESSAGES/lotus.po` and translate the strings as needed.
```
Worth mentioning the locale-specific PO files can be generated using `mix gettext.merge priv/lotus_gettext` 🙂

tugayac commented on README.md:
```diff
@@ -301,6 +302,33 @@ lotus_dashboard "/lotus",
+
+### Providing translations from a host application
+
+1. Create a place for your translations (for example `priv/lotus_gettext` in your host app) and copy the template from `deps/lotus_web/priv/gettext/lotus.pot` as a starting point.
```
You can reference the POT file that will be available once this PR is merged: https://github.com/typhoonworks/lotus_web/blob/main/priv/gettext/lotus.pot (probably easier for users of the library to copy from there instead of the deps directory).

tugayac commented:
Hey @Snunesnk 👋 thank you for this PR, really appreciate your effort to contribute ❤️

I gave your PR a run locally, following the instructions you added to the README, but I wasn't able to get it working. Note, I tested the changes by importing your branch into another Elixir project (similar to how someone would use the Lotus Web library).

<details>
<summary>Here's the Git patch showing how I was testing the PR</summary>

("Accomplish" is the name of the other Elixir project I was using for testing)
```
diff --git a/config/config.exs b/config/config.exs
index 5062c76e..3336687e 100644
--- a/config/config.exs
+++ b/config/config.exs
@@ -184,6 +184,11 @@ config :accomplish, Accomplish.PDFExtractor,
   path: "native/pdf_extractor",
   mode: :debug
 
+config :lotus_web, Lotus.Web.Gettext,
+  priv: "priv/lotus_gettext",
+  locales: ~w(en tr),
+  default_locale: "en"
+
 # Import environment specific config. This must remain at the bottom
 # of this file so it overrides the configuration defined above.
 import_config "#{config_env()}.exs"
diff --git a/lib/accomplish_web/router.ex b/lib/accomplish_web/router.ex
index 951b8e80..85af4ad8 100644
--- a/lib/accomplish_web/router.ex
+++ b/lib/accomplish_web/router.ex
@@ -8,6 +8,12 @@ defmodule AccomplishWeb.Router do
+  def gettext_lotus_plug(conn, _opts) do
+    Gettext.put_locale(Lotus.Web.Gettext, "tr")
+
+    conn
+  end
+
   pipeline :browser do
     plug :accepts, ["html"]
     plug :fetch_session
@@ -43,6 +49,7 @@ defmodule AccomplishWeb.Router do
     plug :browser
     plug :fetch_current_user
     plug :require_admin_user
+    plug :gettext_lotus_plug
   end
 
   scope "/admin", AccomplishWeb.Admin do
diff --git a/mix.exs b/mix.exs
index ffd13aab..e1e36812 100644
--- a/mix.exs
+++ b/mix.exs
@@ -87,7 +87,7 @@ defmodule Accomplish.MixProject do
-      {:lotus_web, "~> 0.6.2"},
+      {:lotus_web, path: "../lotus_web_branch"},
```

</details>

Here's the priv folder setup:
<img width="342" height="248" alt="Screenshot 2025-11-26 at 10 39 10" src="https://github.com/user-attachments/assets/7610e807-3f0d-4735-aeeb-3d5138827626" />

Here's an example of a translation I have within the tr PO file:
```
#: lib/lotus/web/pages/queries_page.ex:18
#, elixir-autogen, elixir-format
msgid "Queries"
msgstr "Sorgular"
```

Given that you've managed to [get it working based on your comment here](https://github.com/typhoonworks/lotus_web/issues/45#issuecomment-3572065605), perhaps I'm doing something wrong? 🤔

Snunesnk commented:
Hi @tugayac :wave: 

Oops, your setup seems correct so I may have overlooked a step in the process. I'll try to replicate it later this day, I'll keep you updated.

Snunesnk commented:
I have updated this PR.

Now Lotus has french translations, and provide them directly to the host app, following `:lotus_locale` value or "en" by default.

I have successfully tested it in my host app, only adding a plug to the router:

```elixir
...
  pipeline :browser do
  ...
    plug :ensure_lotus_locale

...

  defp ensure_lotus_locale(conn, _opts) do
    locale = Plug.Conn.get_session(conn, :lotus_locale) || "fr"
    Plug.Conn.put_session(conn, :lotus_locale, locale)
  end
```

Please tell me if this is also working on your side, and if the PR looks good to you.
I also updated README to reflect those changes, of course feel free to update it if needed.

Thanks for your time !

tugayac commented on lib/lotus/web/gettext.ex:
```diff
@@ -0,0 +1,14 @@
+
+  use Gettext.Backend,
+    otp_app: :lotus_web,
+    default_domain: "lotus"
```
We can keep the domain as `default`, leaves better room for breaking into domains in the future.

tugayac commented on lib/lotus/web/router.ex:
```diff
@@ -113,12 +116,49 @@ defmodule Lotus.Web.Router do
+  defp normalize_locale(locale) when is_atom(locale),
+    do: locale |> Atom.to_string() |> normalize_locale()
+
+  defp normalize_locale(_locale), do: nil
```
Updated in https://github.com/typhoonworks/lotus_web/pull/46/commits/17a625b691e3a59ec49a69f1251bbfabd540651d

tugayac commented on mix.exs:
```diff
@@ -44,7 +44,7 @@ defmodule Lotus.Web.MixProject do
       {:phoenix_html, "~> 4.0"},
       {:phoenix_ecto, "~> 4.5"},
-      {:gettext, "~> 1.0"},
+      {:gettext, "~> 0.26"},
```
I downgraded to 0.26 because it seems a lot of Elixir libraries still use `0.26` as the target version so if we use `~> 1.0` it causes dependency conflicts. Good news is `1.0.0` [doesn't introduce any breaking changes](https://hexdocs.pm/gettext/changelog.html#v1-0-0), so it's safe for folks to use `1.0.0` in their apps if they'd like 🙂.

Snunesnk commented:
Good news !
Yes sure, I'm going to pull your modifications to confirm it's still working :+1:

Snunesnk commented:
I can confirm, it is working just fine !

Thank you again for your time and effort helping me on this PR :pray:
